src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/DefaultRazorLanguageServerCustomMessageTarget.cs

```diff
@@ -625,5 +626,90 @@ public override async Task<VSInternalWrapWithTagResponse> RazorWrapWithTagAsync(
 
             return request?.Response;
         }
+
+        public override async Task<RazorFoldingRangeResponse> ProvideFoldingRangesAsync(RazorFoldingRangeRequestParam foldingRangeParams, CancellationToken cancellationToken)
+        {
+            if (foldingRangeParams is null)
+            {
+                throw new ArgumentNullException(nameof(foldingRangeParams));
+            }
+
+            var csharpRanges = new List<OmniSharp.Extensions.LanguageServer.Protocol.Models.FoldingRange>();
```

Comment:
Things should serialize the same, this can be the VS folding range type since we're on the VS side

Ditto with the HTML variant

Reply:
Ah, so it's fine to just use the VS type here and assume it will deserialize correctly?

Reply:
Yup because that's what the platform will understand too. Aka it's the data type that all VS components will "most likely" share.